libraries/aws_vpc_endpoint.rb

```diff
@@ -0,0 +1,55 @@
+class AWSVpcEndpoint < Inspec.resource(1)
+  name 'aws_vpc_endpoint'
+  desc 'Test the configuration of a VPC endpoint.'
+  example "
+    describe aws_vpc_endpoint('vpce-1234') do
+      it { should exist }
+      its('status') { should eq 'available' }
+      its('vpc_id') { should eq 'vpc-00000' }
+      its('endpoint_type') { should eq 'Interface' }
+      its('service_name') { should eq 'com.amazonaws.ap-southeast-2.s3' }
+      its('route_tables') { should include 'rtb-00000' }
+    end
+  "
+
+  def initialize(vpc_endpoint_id)
+    @vpc_endpoint_id = vpc_endpoint_id
+    @vpc_endpoint = get_vpc_endpoint(vpc_endpoint_id)
+  end
+
+  def to_s
+    "VPC Endpoint #{@vpc_endpoint_id}"
+  end
+
+  def exists?
+    @vpc_endpoint != nil
+  end
+
+  def status
+    @vpc_endpoint.state
+  end
+
+  def vpc_id
+    @vpc_endpoint.vpc_id
+  end
+
+  def endpoint_type
+    @vpc_endpoint.vpc_endpoint_type
+  end
+
+  def service_name
+    @vpc_endpoint.service_name
+  end
+
+  def route_tables
+    @vpc_endpoint.route_table_ids
+  end
+
+  private
+
+  def get_vpc_endpoint(vpc_endpoint_id)
+    ec2 = Aws::EC2::Client.new
+    vpc_endpoints = ec2.describe_vpc_endpoints(vpc_endpoint_ids: [vpc_endpoint_id]).vpc_endpoints
+    vpc_endpoints[0]
```

Comment:
Won't this fail if there's no matching VPC endpoint? (in which case, `exists?` won't pass anyway)

Reply:
It fails but in the control it puts out something like: 
``` 
The Vpc Endpoint Id 'vpce-12345' does not exist
```
I don't think it makes much sense to handle the error as the tests cannot even be evaluated if the VPC endpoint does not exist.